cli/commands/preview/lib/appdata.ts

```diff
@@ -0,0 +1,132 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { z } from 'zod';
+import { LoggerError } from 'cli/logger';
+
+const SnapshotSchema = z.object( {
+	url: z.string(),
+	atomicSiteId: z.number(),
+	localSiteId: z.string(),
+	date: z.number(),
+	name: z.string().optional(),
+	userId: z.number().optional(),
+} );
+
+const UserDataSchema = z.object( {
+	version: z.number().optional(),
+	sites: z.array( z.any() ).optional(),
+	snapshots: z.array( SnapshotSchema ).optional(),
+	authToken: z.any().optional(),
+	locale: z.string().optional(),
+} );
+
+type Snapshot = z.infer< typeof SnapshotSchema >;
+type UserData = z.infer< typeof UserDataSchema >;
+
+// ToDo: Improve this to support multiple platforms
+export function getAppdataPath(): string {
+	const homeDir = os.homedir();
+	return path.join( homeDir, 'Library', 'Application Support', 'Studio', 'appdata-v1.json' );
+}
+
+export async function readAppdata( action: string ): Promise< UserData > {
+	const appDataPath = getAppdataPath();
+
+	if ( ! fs.existsSync( appDataPath ) ) {
+		throw new LoggerError( 'Appdata file not found. Please run the Studio app first.', action );
+	}
+
+	try {
+		const fileContent = fs.readFileSync( appDataPath, 'utf8' );
+		const userData = JSON.parse( fileContent );
+		const result = UserDataSchema.safeParse( userData );
+
+		if ( ! result.success ) {
+			throw new LoggerError( `Invalid appdata format. Please run the Studio app again.`, action );
+		}
```

Comment:
```suggestion
		const result = UserDataSchema.parse( userData );
```
Since we already have an extensive `catch` clause here, it seems like we could use the `zod.parse` method.